bank_classes/account_class.rb

```diff
@@ -47,42 +48,65 @@ def self.find(id)
 # ### Error handling
 # - A new account cannot be created with initial negative balance - this will `raise` an `ArgumentError` (Google this)
   # begin
-    def open_deposit_check(initial_deposit)
-    transacting = true
-    while transacting do
-      if initial_deposit <= 0
-        raise Exception.new("New accounts must be opened with a positive balance.")
-      else
-        transacting = false
+    def open_deposit_check(initial_deposit, open_deposit_min)
+      if initial_deposit <= open_deposit_min
+        raise Exception.new("New accounts must be opened with a balance of at least #{open_deposit_min}.")
       end
       return initial_deposit.to_i
     end
-  end
 
 # - Should have a `withdraw` method that accepts a single parameter which represents the amount of money that will be withdrawn. This method should return the updated account balance.
-    def withdraw (withdraw_amount)
+    def withdraw (withdraw_amount, fee = 0, min_balance = 0)
```

Comment:
If a user calls this method will they be specifying their own fee amount? I'd choose zero :+1: